gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/SlackController.java

```diff
@@ -62,33 +70,79 @@ public SlackController(
     return slackChannelsCache.get();
   }
 
-  @Scheduled(
-      fixedDelayString = "${slack.channel-refresh-interval-millis:1200000}",
-      initialDelayString = "${random.int(600000)}")
+  @Scheduled(fixedDelayString = "${slack.channel-refresh-interval-millis:600000}")
   void refreshSlack() {
     try {
-      log.info("Refreshing Slack channels list");
-      List<Map> channels = fetchChannels();
-      log.info("Fetched {} Slack channels", channels.size());
-      slackChannelsCache.set(channels);
+      Long startTime = System.nanoTime();
+      log.info("Refreshing Slack channels");
+      slackChannelsCache.set(fetchChannels());
+      log.info(
+          "Fetched {} Slack channels in {}ms",
+          slackChannelsCache.get().size(),
+          TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startTime));
+      slackChannelsCacheLastRefreshed = new Date();
     } catch (Exception e) {
       registry.counter("slack.channels.errors").increment();
-      log.error("Unable to refresh Slack channels list", e);
+      log.error(
+          "Unable to refresh Slack channels (last successful: {})",
+          slackChannelsCacheLastRefreshed,
+          e);
     }
   }
 
-  List<Map> fetchChannels() {
-    SlackService.SlackChannelsResult response =
-        AuthenticatedRequest.allowAnonymous(
-            () -> slackService.getChannels(slackConfigProperties.getToken(), null));
+  private List<Map> fetchChannels() {
+    SlackService.SlackChannelsResult response = getChannels(slackConfigProperties.getToken(), null);
     List<Map> channels = response.channels;
     String cursor = response.response_metadata.next_cursor;
-    while (cursor != null & cursor.length() > 0) {
-      response = slackService.getChannels(slackConfigProperties.getToken(), cursor);
+    while (cursor != null && cursor.length() > 0) {
+      response = getChannels(slackConfigProperties.getToken(), cursor);
       cursor = response.response_metadata.next_cursor;
       channels.addAll(response.channels);
     }
 
     return channels;
   }
+
+  private SlackService.SlackChannelsResult getChannels(String token, String cursor) {
+    return retrySupport.retry(
+        () -> {
+          try {
+            return slackService.getChannels(token, cursor);
```

Comment:
Does `AuthenticatedRequest.allowAnonymous` still belong here to keep from generating log spam?

Reply:
We've been running this in our environment with no logging spam.